internal/providers/terraform/azure/notification_hubs.go

```diff
@@ -0,0 +1,102 @@
+package azure
+
+import (
+	"fmt"
+
+	"github.com/infracost/infracost/internal/schema"
+	"github.com/shopspring/decimal"
+	"github.com/tidwall/gjson"
+)
+
+func GetAzureRMNotificationHubsRegistryItem() *schema.RegistryItem {
+	return &schema.RegistryItem{
+		Name:  "azurerm_notification_hub_namespace",
+		RFunc: NewAzureRMNotificationHubs,
+	}
+}
+
+func NewAzureRMNotificationHubs(d *schema.ResourceData, u *schema.UsageData) *schema.Resource {
+	var monthlyAdditionalPushes *decimal.Decimal
+	var startUsageAmt string
+
+	if u != nil && u.Get("monthly_additional_pushes").Type != gjson.Null {
+		monthlyAdditionalPushes = decimalPtr(decimal.NewFromInt(u.Get("monthly_additional_pushes").Int()))
+	}
+
+	sku := "Basic"
+	location := d.Get("location").String()
+
+	if d.Get("sku_name").Type != gjson.Null {
+		sku = d.Get("sku_name").String()
+	}
+	costComponents := make([]*schema.CostComponent, 0)
+
+	costComponents = append(costComponents, notificationHubsCostComponent("Base Charge Per Namespace", location, sku))
+	if monthlyAdditionalPushes.GreaterThan(decimal.NewFromInt(10000000)) {
```

Comment:
You can use the usage.CalculateTierBuckets helper for this.  Take a look at https://github.com/infracost/infracost/blob/master/internal/providers/terraform/azure/key_vault_key.go#L90-L102 for an example.